### PR TITLE
Sprint/1

### PR DESCRIPTION
**Pull request by SebastianDaszewski**

### CODE REVIEW

**Comment by SebastianDaszewski** [comment]
Can you please tell me what I exactly need to change? Background color in social media bar? I can't see more differences on my screen.

**Comment by SebastianDaszewski** [comment]
Please take a look now.

**Review by elstyropiano — CHANGES_REQUESTED** [review]
Remember to change ci.yml file becasuse now we have build error:
image
Rafał explained how to do that.

Please add also package like we talk on slack:
image
Please remember this site should be responsive from 1024 px
image

image
image
image
image


- Please apply styles in your project so that they are consistent with the graphic design. You can trace point by point whether all elements on the page have been included and whether the size spacing and colors are correct

**Comment by elstyropiano** [comment]
remember to use clsx library in case like that 
image
It was was included in docs https://trello.com/c/rxvtwg22/6-docs

If you are not use some variables or props like here, please don't leave like that, just remove it :
image

File: app/[locale]/login/page.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import "../globals.css";
+import Image from "next/image";
+import Header from "@/features/home/components/Header";
+import Footer from "@/features/home/components/Footer";
+import Content from "@/features/home/components/Content";
+import LoginForm from "@/features/home/components/LoginForm";
+import ConfirmModal from "@/features/home/components/ConfirmModal";
+import { useTranslations } from "next-intl";
+
+export default function Page() {
```
Comment (elstyropiano) [inline app/[locale]/login/page.tsx:11]: Please use only arrow functions, move export also in down below component. `Page=> LoginPage`.According to the documentation you should return here </Login> , correct also it in registration and landing(after create it) routes

File: app/[locale]/login/page.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+          password={l("password")}
+          email={l("email")}
+          signIn={l("signIn")}
+          githubLogin={l("githubLogin")}
+          rememberMe={l("rememberMe")}
+          forgotPassword={l("forgotPassword")}
+          stillDontHaveAccount={l("stillDontHaveAccount")}
+          registration={l("registration")}
+          passwordRequired={l("passwordRequired")}
+          wrongEmail={l("wrongEmail")}
```
Comment (elstyropiano) [inline app/[locale]/login/page.tsx:33]: Do you need pass props here ?

File: app/[locale]/login/page.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import Header from "@/features/home/components/Header";
+import Footer from "@/features/home/components/Footer";
+import Content from "@/features/home/components/Content";
+import LoginForm from "@/features/home/components/LoginForm";
+import ConfirmModal from "@/features/home/components/ConfirmModal";
+import { useTranslations } from "next-intl";
+
+export default function Page() {
+  const t = useTranslations("TopBar");
+  const l = useTranslations("loginForm");
```
Comment (elstyropiano) [inline app/[locale]/login/page.tsx:13]: Please use only one translation for file

File: app/[locale]/login/page.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  return (
+    <>
```
Comment (elstyropiano) [inline app/[locale]/login/page.tsx:16]: Do we need that react fragment ?

File: app/[locale]/page.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  const router = useRouter();
+
+  return (
+    <div className="bg-black flex flex-col items-center justify-center">
+      <Header
+        signIn={t("signIn")}
+        registrationButtonColor="dark:bg-blue-600 bg-blue-600"
+        loginButtonColor="dark:bg-gray-900"
+      />
+
```
Comment (elstyropiano) [inline app/[locale]/page.tsx:23]: Please don't leave spaces in jsx

File: app/[locale]/login/page.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+          email={l("email")}
+          signIn={l("signIn")}
+          githubLogin={l("githubLogin")}
+          rememberMe={l("rememberMe")}
+          forgotPassword={l("forgotPassword")}
+          stillDontHaveAccount={l("stillDontHaveAccount")}
+          registration={l("registration")}
+          passwordRequired={l("passwordRequired")}
+          wrongEmail={l("wrongEmail")}
+        />{" "}
```
Comment (elstyropiano) [inline app/[locale]/login/page.tsx:34]: Please dont leave elements like that `{" "}`. Aplly it to all files

File: app/[locale]/registration/page.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+export default function Page() {
+  const t = useTranslations("TopBar");
+  const r = useTranslations("registrationForm");
+
+  return (
+    <>
+      <div className="bg-black flex flex-col items-center justify-center">
+        <Header signIn={t("signIn")} registrationButtonColor="bg-yellow-500" />
+        <RegistrationForm
```
Comment (elstyropiano) [inline app/[locale]/registration/page.tsx:19]: I think u don't need pass props like that

File: features/home/components/.Header.tsx.swp
Comment (elstyropiano) [inline features/home/components/.Header.tsx.swp:1]: Do you need that file  ?

File: features/home/components/ConfirmModal.tsx [hunk 8]
```diff
@@ -0,0 +1,31 @@
+"use client";
+import React from "react";
```
Comment (elstyropiano) [inline features/home/components/ConfirmModal.tsx:2]: I think you don't need that import

File: features/home/components/ConfirmModal.tsx [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+const ConfirmModal = () => {
+  return (
+    <>
```
Comment (elstyropiano) [inline features/home/components/ConfirmModal.tsx:6]: I think you don't need that

File: features/home/components/Content.tsx [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+const Content = ({
+  texts,
+  contentTitle,
+  heroText,
+  startLearningBtn,
+  getToKnowBtn,
+}: Props) => {
+  const isTextPresent =
+    texts && Object.values(texts).some((text) => text.length > 0);
+  const shouldShowImageOnLeft = !isTextPresent;
```
Comment (elstyropiano) [inline features/home/components/Content.tsx:32]: something is wrong here if is true
image

File: features/home/components/Content.tsx [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+                  {
+                    hidden: shouldShowImageOnLeft === true,
+                  }
+                )}
+              >
+                {texts?.text1}
+              </h6>
+              <h6
+                className={clsx(
+                  "font-lexend-deca tracking-wider text-black w-520 font-thin 1024:leading-24px  1024:w-400 1024:text-14px 1366:w-520 1366:mt-5 1366:text-20px 1366:leading-30px ",
```
Comment (elstyropiano) [inline features/home/components/Content.tsx:122]: Spaces also, please check your files and correct.

File: features/home/components/Content.tsx [hunk 12]
```diff
@@ -0,0 +1,189 @@
...
+
+type Texts = {
+  text1: string;
+  text2: string;
+  text3: string;
+  text4: string;
+  text5: string;
+};
+
+type Props = {
```
Comment (elstyropiano) [inline features/home/components/Content.tsx:15]: Please check docs and write how to names type

**Review by elstyropiano — CHANGES_REQUESTED** [review]
It's looks better than before but we have to correct some places :

- Please use `font-family: Lexend Deca;`
- padding should equal 40 px, also in social media, now is not in one line 

image
image

- Please correct modal

image

- something is wrong with that place on big screen, footer should be on the bottom, the same in another routes:

image

- When we go to the login or registration page, many errors appear. So please fix that 
image

- we have  no /landing route :
image

image

- Please write docks , and correct files 
image

- When i click on submit button, errors should appear : 
image. But now we have default. In registration form also

File: app/[locale]/login/page.tsx (hunk 4 above)
Comment (elstyropiano) [reply app/[locale]/login/page.tsx:9]: Please follow the previous advice

**Review by elstyropiano — CHANGES_REQUESTED** [review]
Looks better but please apply all advices from previous review: 
image
image
image

- Button should be higher

image

**Comment by elstyropiano** [comment]
Please resolve that also :
image

**Comment by elstyropiano** [comment]
And please resolve build errors 
image

File: app/[locale]/landing/page.tsx [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+
+import { useTranslations } from "next-intl";
+
+import Header from "@/features/home/components/Header";
+import Footer from "@/features/home/components/Footer";
+import Content from "@/features/home/components/Content";
+import SocialMediaBar from "@/features/home/components/SocialMediaBar";
+import TopBar from "@/features/home/components/TopBar";
+
+import "./../globals.css";
```
Comment (elstyropiano) [inline app/[locale]/landing/page.tsx:11]: please remove

File: features/home/components/Content.tsx [hunk 14]
```diff
@@ -0,0 +1,179 @@
+"use client";
+
+import Background from "./Background";
+import clsx from "clsx";
+import { useTranslations } from "next-intl";
+
+import Image from "next/legacy/image";
```
Comment (elstyropiano) [inline features/home/components/Content.tsx:7]: import orders

File: features/home/components/LoginForm.tsx [hunk 15]
```diff
@@ -0,0 +1,152 @@
...
+
+import { useTranslations } from "next-intl";
+import clsx from "clsx";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { z } from "zod";
+import React, { useState } from "react";
+import Image from "next/legacy/image";
+
+import { zodResolver } from "@hookform/resolvers/zod";
+import Background from "./Background";
```
Comment (elstyropiano) [inline features/home/components/LoginForm.tsx:11]: import orders

File: features/home/components/LoginForm.tsx [hunk 16]
```diff
@@ -0,0 +1,152 @@
...
+              className="text-sm/5.25 font-medium text-white"
+              htmlFor="password"
+            >
+              {t("password")}
+            </label>
+            <input
+              type="password"
+              id="password"
+              className={clsx(
+                "w-96 h-10.5 bg-gray-700 border border-gray-600 text-gray-900 text-sm rounded-lg py-3	px-4",
```
Comment (elstyropiano) [inline features/home/components/LoginForm.tsx:84]: please remove space

File: features/home/components/LoginForm.tsx [hunk 17]
```diff
@@ -0,0 +1,152 @@
...
+            <label
+              htmlFor="terms"
+              className="mt-3 ml-3 text-sm font-normal/3.5 text-white"
+            >
+              {t("rememberMe")}
+            </label>
+          </div>
+          <button
+            type="submit"
+            className="mt-7 mb-1 font-medium w-96 h-10 py-2.5	px-5 text-white bg-blue-700 rounded-lg"
```
Comment (elstyropiano) [inline features/home/components/LoginForm.tsx:115]: same as above

File: features/home/components/RegistrationForm.tsx [hunk 18]
```diff
@@ -0,0 +1,309 @@
...
+
+import React, { useState } from "react";
+import Image from "next/legacy/image";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { z } from "zod";
+import clsx from "clsx";
+import { useTranslations } from "next-intl";
+
+import { zodResolver } from "@hookform/resolvers/zod";
+import Background from "./Background";
```
Comment (elstyropiano) [inline features/home/components/RegistrationForm.tsx:11]: import orders

File: features/home/components/RegistrationForm.tsx [hunk 19]
```diff
@@ -0,0 +1,309 @@
...
+                      className="text-sm/5.25 font-medium text-white"
+                      htmlFor="pseudonim"
+                    >
+                      {t("nick")}
+                    </label>
+                    <input
+                      type="text"
+                      id="pseudonim"
+                      className={clsx(
+                        "h-10.5 w-76.25 bg-gray-700 border border-gray-600 text-white text-sm leading-4.375 font-normal rounded-lg focus:ring-blue-500 focus:border-blue-500 py-3	px-4",
```
Comment (elstyropiano) [inline features/home/components/RegistrationForm.tsx:95]: please remove spaces

File: features/home/components/RegistrationForm.tsx [hunk 20]
```diff
@@ -0,0 +1,309 @@
...
+                      className="text-sm/5.25 font-medium text-white"
+                      htmlFor="firstName"
+                    >
+                      {t("name")}
+                    </label>
+                    <input
+                      type="text"
+                      id="firstName"
+                      className={clsx(
+                        "h-10.5 w-76.25 bg-gray-700 border border-gray-600 text-white text-sm/4.375 font-normal rounded-lg focus:ring-blue-500 focus:border-blue-500 py-3	px-4",
```
Comment (elstyropiano) [inline features/home/components/RegistrationForm.tsx:145]: same here

File: features/home/components/RegistrationForm.tsx [hunk 21]
```diff
@@ -0,0 +1,309 @@
...
+                  className="mt-7 text-sm/5.25 font-medium text-white"
+                  htmlFor="password"
+                >
+                  {t("password")}
+                </label>
+                <input
+                  type="password"
+                  id="password"
+                  className={clsx(
+                    "h-10.5 w-159 bg-gray-700 border border-gray-600 text-white text-sm/4.375 rounded-lg focus:ring-blue-500 focus:border-blue-500 block py-3	px-4",
```
Comment (elstyropiano) [inline features/home/components/RegistrationForm.tsx:196]: here also

File: features/home/components/RegistrationForm.tsx [hunk 22]
```diff
@@ -0,0 +1,309 @@
...
+                  className="text-sm/5.25 font-medium text-white"
+                  htmlFor="confirm_password"
+                >
+                  {t("confirmPassword")}
+                </label>
+                <input
+                  type="password"
+                  id="confirmPassword"
+                  className={clsx(
+                    "h-10.5 w-159 bg-gray-700 border border-gray-600 text-white text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 py-3	px-4",
```
Comment (elstyropiano) [inline features/home/components/RegistrationForm.tsx:221]: here also

File: features/home/components/TopBar.tsx [hunk 23]
```diff
@@ -0,0 +1,56 @@
+"use client";
+
+import { useTranslations } from "next-intl";
+
+import Image from "next/image";
```
Comment (elstyropiano) [inline features/home/components/TopBar.tsx:5]: import orders

File: tailwind.config.js [hunk 24]
```diff
@@ -10,11 +10,114 @@ module.exports = {
...
+        163.5: "40.875rem",
+        175: "43.75rem",
+        198.5: "49.625rem",
+        200: "50rem",
+        218.5: "54.625rem",
+        192: "48rem",
+        212.5: "53.125rem",
+        341.5: "85.375rem",
+        500: "125rem",
+      },
```
Comment (elstyropiano) [inline tailwind.config.js:120]: please remove unecessary sizes

**Review by elstyropiano — CHANGES_REQUESTED** [review]
Once again ask you to follow the instructions and draw conclusions after the review

**Comment by elstyropiano** [comment]
image
Please don't use square brackets in style

**Comment by elstyropiano** [comment]
- Please take this into consideration as well

image

**Review by elstyropiano — APPROVED** [review]

File: features/landing/components/Landing.tsx [hunk 25]
```diff
@@ -0,0 +1,35 @@
+"use client";
+
+import { useTranslations } from "next-intl";
+
+import Header from "@/components/Header";
+import Footer from "@/components/Footer";
+import Content from "@/features/landing/components/Content";
+
+const Landing = () => {
+  const l = useTranslations("Content");
```
Comment (elstyropiano) [inline features/landing/components/Landing.tsx:10]: `l=>t`

File: features/login/components/LoginForm.tsx [hunk 26]
```diff
@@ -0,0 +1,153 @@
+"use client";
+
+import { useTranslations } from "next-intl";
+import clsx from "clsx";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { z } from "zod";
+import React, { useState } from "react";
+import Image from "next/legacy/image";
+
+import { zodResolver } from "@hookform/resolvers/zod";
```
Comment (elstyropiano) [inline features/login/components/LoginForm.tsx:10]: its import from package

File: features/register/components/RegistrationForm.tsx [hunk 27]
```diff
@@ -0,0 +1,308 @@
+"use client";
+
+import { useState } from "react";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { z } from "zod";
+import clsx from "clsx";
+import { useTranslations } from "next-intl";
+
+import { zodResolver } from "@hookform/resolvers/zod";
```
Comment (elstyropiano) [inline features/register/components/RegistrationForm.tsx:9]: same here

File: features/register/components/RegistrationForm.tsx [hunk 28]
```diff
@@ -0,0 +1,308 @@
...
+                      className="text-sm/5.25 font-medium text-white"
+                      htmlFor="firstName"
+                    >
+                      {t("name")}
+                    </label>
+                    <input
+                      type="text"
+                      id="firstName"
+                      className={clsx(
+                        "h-10.5 w-76.25 bg-gray-700 border border-gray-600 text-white text-sm/4.375 font-normal rounded-lg focus:ring-blue-500 focus:border-blue-500 py-3	px-4",
```
Comment (elstyropiano) [inline features/register/components/RegistrationForm.tsx:144]: ???
image

File: features/register/components/RegistrationForm.tsx [hunk 29]
```diff
@@ -0,0 +1,308 @@
...
+                      className="text-sm/5.25 font-medium text-white"
+                      htmlFor="pseudonim"
+                    >
+                      {t("nick")}
+                    </label>
+                    <input
+                      type="text"
+                      id="pseudonim"
+                      className={clsx(
+                        "h-10.5 w-76.25 bg-gray-700 border border-gray-600 text-white text-sm leading-4.375 font-normal rounded-lg focus:ring-blue-500 focus:border-blue-500 py-3	px-4",
```
Comment (elstyropiano) [inline features/register/components/RegistrationForm.tsx:94]: image
???

File: features/register/components/RegistrationForm.tsx [hunk 30]
```diff
@@ -0,0 +1,308 @@
...
+        <div className="relative flex justify-center w-screen h-full">
+          <Background />
+          <div className="top-50 tall:top-100 w-175 h-163.5 p-9 rounded-lg gap-32 text-white flex flex-col justify-start items-center relative shadow-custom bg-customGray shortMax:top-0 shortMax:scale-90">
+            <form className="w-159" onSubmit={handleSubmit(onSubmit)}>
+              <span className="text-2xl/9 font-extralight text-white">
+                {t("registration")}
+              </span>
+              <div className="flex mt-10">
+                <div className="mr-6">
+                  <div className="mb-7">
```
Comment (elstyropiano) [inline features/register/components/RegistrationForm.tsx:83]: What do you think to create reusable component for input ? and use it also in login form its not sense to create every time new input with the same props

File: features/register/components/RegistrationForm.tsx [hunk 31]
```diff
@@ -0,0 +1,308 @@
...
+                  placeholder="•••••••••"
+                  {...register("confirmPassword")}
+                />
+                {errors.confirmPassword && (
+                  <p className="absolute text-xs/4.5 font-extralight text-red-500">
+                    {errors.confirmPassword.message}
+                  </p>
+                )}
+              </div>
+
```
Comment (elstyropiano) [inline features/register/components/RegistrationForm.tsx:234]: ??

File: features/register/components/RegistrationForm.tsx [hunk 32]
```diff
@@ -0,0 +1,308 @@
...
+                  className="text-sm/5.25 font-medium text-white"
+                  htmlFor="confirm_password"
+                >
+                  {t("confirmPassword")}
+                </label>
+                <input
+                  type="password"
+                  id="confirmPassword"
+                  className={clsx(
+                    "h-10.5 w-159 bg-gray-700 border border-gray-600 text-white text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 py-3	px-4",
```
Comment (elstyropiano) [inline features/register/components/RegistrationForm.tsx:220]: ??

File: features/register/components/RegistrationForm.tsx [hunk 33]
```diff
@@ -0,0 +1,308 @@
...
+                {errors.confirmPassword && (
+                  <p className="absolute text-xs/4.5 font-extralight text-red-500">
+                    {errors.confirmPassword.message}
+                  </p>
+                )}
+              </div>
+
+              <div className="flex items-start mb-6 mt-5">
+                <div className="flex items-center">
+                  <input
```
Comment (elstyropiano) [inline features/register/components/RegistrationForm.tsx:237]: Consider about create component for checkbox also

File: tailwind.config.js [hunk 34]
```diff
@@ -10,11 +10,92 @@ module.exports = {
...
+        20: "5rem",
+        49.5: "12.375rem",
+        50: "12.5rem",
+        69.5: "17.375rem",
+        100: "25rem",
+        120: "30rem",
+        141: "35.25rem",
+        150: "37.5rem",
+        163.5: "40.875rem",
+        175: "43.75rem",
```
Comment (elstyropiano) [inline tailwind.config.js:97]: Please check tailwind docks andremove the styles provided by tailwind

**Review by elstyropiano — CHANGES_REQUESTED** [review]
image

File: features/landing/components/Landing.tsx [hunk 35]
```diff
@@ -0,0 +1,35 @@
+"use client";
+
+import { useTranslations } from "next-intl";
+
+import { Header } from "@/components";
+import { Footer } from "@/components";
+import Content from "@/features/landing/components/Content";
```
Comment (elstyropiano) [inline features/landing/components/Landing.tsx:7]: Wy import like that ?

File: features/landing/components/Landing.tsx [hunk 36]
```diff
@@ -0,0 +1,35 @@
+"use client";
+
+import { useTranslations } from "next-intl";
+
+import { Header } from "@/components";
+import { Footer } from "@/components";
```
Comment (elstyropiano) [inline features/landing/components/Landing.tsx:6]: You can write it in one line , but i think you don;t need that here. Better to make some shared layout for features beacasue in login and in register code  you import Header and footer also

File: components/SocialMediaBar.tsx [hunk 37]
```diff
@@ -0,0 +1,57 @@
...
+      src: "/images/Tiktok.jpg",
+      alt: "Tiktok",
+    },
+  ];
+
+  return (
+    <div className="flex h-10 items-center bg-black">
+      <div className="w-screen h-10 flex justify-end items-center gap-8 ml-10 mr-10">
+        {socialMediaLinks.map((media, index) => (
+          <a
```
Comment (elstyropiano) [inline components/SocialMediaBar.tsx:43]: Please Use Link from next and everywhere where you have <a>

**Review by elstyropiano — CHANGES_REQUESTED** [review]
- What about it what im wrote in previous review ?

image

- Please use svg icons fromat, not jpg

**Review by elstyropiano — CHANGES_REQUESTED** [review]
- Still we have problem here, if we are on landing route

image

- Please remove font-lexend-deca from class like here 
image
Beacause you defined it in parent 
image

**Review by elstyropiano — APPROVED** [review]

**Pull request merged**